pkg/cluster/nodeutils/roles.go

```diff
@@ -66,6 +66,8 @@ func ExternalLoadBalancerNode(allNodes []nodes.Node) (nodes.Node, error) {
 	return loadBalancerNodes[0], nil
 }
 
+// APIServerEndpointNode try to get api server endpoint node
```

Comment:
```suggestion
// APIServerEndpointNode selects the node from allNodes which hosts the API Server endpoint
// This should be the control plane node if there is one control plane node, or a LoadBalancer otherwise.
// It returns an error if the node list is invalid (E.G. two control planes and no load balancer)
```

Reply:
done